text/0000-language-semver.md

```diff
@@ -0,0 +1,494 @@
+compiler, but this is expected to be straightforward: if supporting
+older behavior is hard to do, it may indicate that the opt-in change
+is too complex and should not be accepted.
+
+Note that the supplied version may affect the parser configuration
+used when parsing the initial crate, since it can affect the keywords
+recognized by the tokenizer and perhaps other minor details in the
+syntax. However, because the version is supplied on the command line,
+this configuration is known before parsing begins.
+
+#### Defaults and extreme cases
+
+If no version is supplied on the `rustc` command line, `rustc` will
+default to the maximal version it recognizes. If the user supplies a
+version `X.Y` that is *newer* than the compiler itself, the compiler
+should simply issue a warning and proceed as if the user had supplied
+the compiler's version (i.e., the newest version the compiler knows
+about).
+
+Cargo will always invoke `rustc` with a supplied version. If there is
+no version in the `Cargo.toml` file, then `1.0.0` is assumed. Whenever
+a new project is created with `cargo new`, the new `Cargo.toml` will
+include the most recent Rust version number by default.
+
+Note that the defaults for `rustc` and `cargo` differ. `rustc` prefers
+the most recent verison of Rust by default, whereas `cargo` prefers
+the oldest. The reason is that we expect running `rustc` in a
+standalone fashion to be used primarily when experimenting with small
+scripts and one-offs, and the user is most likely to want "current
+Rust" in that scenario.
+
+#### When opt-in changes are appropriate
+
+Opt-in changes allow us to greatly expand the scope of the kinds of
+additions we can make without breaking existing code, but they are not
+applicable in all situations. A good rule of thumb is that an opt-in
+change is only appropriate if the exact effect of the older code can
+be easily recreated in the newer system with only surface changes to
+the syntax.
+
+Another view is that opt-in changes are appropriate if those changes
+do not affect the "abstract AST" of your Rust program. In other words,
+existing Rust syntax is just a serialization of a more idealized view
+of the syntax, in which there are no conflicts between keywords and
+identifiers, syntactic sugar is expanded, and so forth. Opt-in changes
+might affect the translation into this abstract AST, but should not
+affect the semantics of the AST itself at a deeper level. This concept
+of an idealized AST is analagous to the "elaborated syntax" described
+in [RFC 1105], except that it is at a conceptual level.
+
+So, for example, the conflict between new keywords and existing
+identifiers can (generally) be trivially worked around by renaming
+identifiers, though the question of public identifiers is an
+interesting one (contextual keywords may suffice, or else perhaps some
+kind of escaping syntax -- we defer this question here for a later
+RFC).
```

Comment:
Another useful example you might include here is if we add a new feature (with a new keyword attached) that adds somethign fundamental to the AST, but where the semantics of code in the absence of that keyword has some "obvious default" meaning with respect to the new class of abstract ASTs of Rust.

As a concrete example of this: if we start allowing explicit lifetimes on arbitrary blocks: `'a: { ... }`, then we might say that Rust's AST has a lifetime on every block, but then blocks written without explicit lifetime simply receive the default of a lifetime-inference variable. 

(Or maybe this doesn't make anything clearer at all, eh...)